app/models/internal/impl/DefaultAlertEvaluationResult.java

```diff
@@ -0,0 +1,98 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package models.internal.impl;
+
+import com.arpnetworking.commons.builder.OvalBuilder;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
+import com.google.common.base.MoreObjects;
+import com.google.common.base.Objects;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
+import models.internal.AlertEvaluationResult;
+
+import java.util.List;
+import java.util.Map;
+
+/**
+ * Default implementation for an {@code AlertEvaluationResult}.
+ *
+ * @author Christian Briones (cbriones at dropbox dot com)
+ */
+@JsonDeserialize(builder = DefaultAlertEvaluationResult.Builder.class)
+public final class DefaultAlertEvaluationResult implements AlertEvaluationResult {
+    private final ImmutableList<ImmutableMap<String, String>> _firingTags;
+
+    private DefaultAlertEvaluationResult(final Builder builder) {
+        _firingTags = builder._firingTags;
+    }
+
+    @Override
+    public ImmutableList<ImmutableMap<String, String>> getFiringTags() {
+        return _firingTags;
+    }
+
+    @Override
+    public boolean equals(final Object o) {
+        if (this == o) {
+            return true;
+        }
+        if (o == null || getClass() != o.getClass()) {
+            return false;
+        }
+        final DefaultAlertEvaluationResult that = (DefaultAlertEvaluationResult) o;
+        return Objects.equal(_firingTags, that._firingTags);
+    }
+
+    @Override
+    public int hashCode() {
+        return Objects.hashCode(_firingTags);
+    }
+
+    @Override
+    public String toString() {
+        return MoreObjects.toStringHelper(this)
+                .add("firingTags", _firingTags)
+                .toString();
+    }
+
+    /**
+     * Builder class for instances of {@code DefaultAlertEvaluationResult}.
+     */
+    public static class Builder extends OvalBuilder<DefaultAlertEvaluationResult> {
+        private ImmutableList<ImmutableMap<String, String>> _firingTags = ImmutableList.of();
+
+        /**
+         * Default constructor for an empty builder.
+         */
+        public Builder() {
+            super(DefaultAlertEvaluationResult::new);
+        }
+
+        /**
+         * Set the firing tags. Defaults to empty.
+         *
+         * @param firingTags The set of firing tags.
+         * @return This instance of {@code Builder}.
+         */
+        @JsonProperty("firingTags")
```

Comment:
This definitely should not be necessary. If it is, combined with the above, I suspect we may be using the wrong `ObjectMapper` instance (e.g. one that is not configured to support our usual builder pattern).